### DISCUSSION TITLE
Shopify workflow (Parcel v2)

### DISCUSSION THREAD

**Discussion by modulareverything** [parcel-bundler/parcel#5497] [Q&A]
I'm using Parcel with Shopify but I don't think my workflow is the most optimal, and I was wondering if there were steps I could take to improve it?

Currently, I have these scripts:

```
  "scripts": {
    "start": "concurrently \"theme watch --allow-live\" \"parcel serve ./src/js/app.js  --dist-dir ./assets\"",
    "build": "parcel build ./src/js/app.js --dist-dir ./assets",
  },
```

The `start` command will successfully watch app.js for changes and output them to the assets folder where Shopify's theme.liquid has it set as an include. Inside the app.js file is a simple console log, and an import to a test SCSS file. The SCSS file is building out to `app.css` which is good, and the js file is building out to `app.js`

The issue is around live reloading. Parcel spins up a local server at localhost:1234 but clicking on it gives me a 404 page

So what is the most optimal Shopify workflow with Parcel which would allow me to use the live reloading?

**Comment by mischnic** [comment]
> Parcel spins up a local server at localhost:1234 but clicking on it gives me a 404 page

That's because it tries to server the an html file, but there is none known to Parcel in your case. `localhost:1234/app.js` should work though.
You could use `parcel watch` instead to only the Parcel HMR server but not the HTTP server.

Do you get an error during HMR updates? 
Which Parcel version are you using?

**Comment by modulareverything** [comment]
Parcel version is
`"parcel": "^2.0.0-beta.1",`

I'm not getting any errors though, in fact it seems to work pretty nicely just using `watch` alone but I feel like the live reloading / HMR aspect is a big missing piece of the puzzle

I guess we're going beyond the realms of Parcel here potentially, but is this something that Parcel can handle somehow? I tried pointing Parcel at Shopify's theme.liquid file but it didn't like that, gave me an error about not understanding the extension which I guess makes sense!

**Reply by mischnic** [reply]
HMR should work with your setup. So you don't get any updates or errors?

**Comment by lqze** [comment]
@chrish-d You should use browser sync to handle the browser refreshing, combining the `theme watch --notify=/tmp/someFile` command from `themekit`. Browser sync can be configured to watch for that file too, so it knows when you make an update. If you update your JS / CSS, then your theme will update, triggering that file, triggering browser-sync. Browser sync should also proxy to your theme URL.

Here's how I have configured my `package.json` scripts for Shopify theme development.
```
"watch-scripts": "parcel watch 'src/scripts/index.js' --no-source-maps --dist-dir './assets'",
"watch-styles": "parcel watch 'src/styles/index.scss' --no-source-maps --dist-dir './assets'",
"watch-theme": "shopify-themekit watch --config './config.development.yml' --vars './.env' --notify='./themeWatchNotify'",
"dev": "concurrently \"npm:watch-*\" \"browser-sync start --config ./bs-config.js\""
```

**Reply by modulareverything** [reply]
Really late reply on this as I didn't have time to come back to on this project, but thanks for sharing!

Would you mind sharing your `themeWatchNotify` and `bs-config.js` files? Just for pure clarity :) I'm not sure I fully understand how `themeWatchNotify` is being used here

**Reply by lqze** [reply]
`themeWatchNotify` is just an arbitrary file that you set browser-sync to watch for.
If you tell themekit to watch that file too, everytime you make an update it will "touch" it and then Browser sync will also recognise that.

**Reply by modulareverything** [reply]
Okay, that workflow makes sense now. So you're not watching the CSS or JS files for a change, but watching to see when `themeWatchNotify` is updated to trigger a refresh?

I get some errors about `Copy the following snippet into your website, just before the closing </body> tag` and also that the port `3001` is currently in use... you wouldn't think that this would be so tricky!

Any suggestions?

Thanks for the quick reply
